Feat: 로그인 성공 시 JWT 토큰 반환 + 토큰 인증 구현

owel-dev opened the pull request:
### 연결된 이슈
[노션 링크](https://usanglee.notion.site/JWT-25b7401177e745c9a13d71bec4e6042e)

## Pull request 설명
Spring Security 이용하여
사용자의 로그인 성공 시 JWT토큰을 반환하고
인증이 필요한 API 호출 시 사용자의 토큰 보유 유무와 토큰이 유효한 토큰인지를 검사하는 기능을 구현하였습니다.

## Test 방법
1. 브랜치 다운로드
    - `git remote update`
    - `git checkout -t origin/feat/백엔드-JWT-토큰-구현`

2. Spring boot server실행.
3. 포스트맨 이용하여 테스트
    - POST /auth 로 회원가입. 
    - POST /auth/login 으로 로그인.

Commit:
feat: 로그인 성공 시 JWT 토큰 반환 구현

Co-authored-by: dlwndms0812 <dlwndms0812@naver.com>

Commit:
feat: 인증이 필요한 api에 요청시 토큰 검사 구현

Co-authored-by: dlwndms0812 <dlwndms0812@naver.com>

dlwndms0812 commented on backend/demo/src/main/java/com/snacks/demo/jwt/JwtProperties.java:
```diff
@@ -0,0 +1,9 @@
+package com.snacks.demo.jwt;
+
+public interface JwtProperties {
```
이거 들어가면 안되는데 gitignore 체크하고, secret 키 변경해야할 것같아요

dlwndms0812 commented:
저기 위에 부분 말고도 jwtautorization 파일에 반영이 안된 부분 하나가 있었는데,(해결 된 것 확인)
 일단 merge 해주시면 제가 jwtproperties랑 필요없는 주석 삭제 하도록 하겠습니다

pr대신 날려주셔서 감사해요!!

dlwndms0812 commented on backend/demo/src/main/java/com/snacks/demo/jwt/JwtAuthorizationFilter.java:
```diff
@@ -0,0 +1,68 @@
+    if (header == null || !header.startsWith(JwtProperties.TOKEN_PREFIX)) {
+      chain.doFilter(request, response);
+//      System.out.println("@ hello" + request.getPathInfo());
+      return;
```
본문에서 누락된것 같다는 부분이었는데, 확인해보니 여기에 있네요!!

dlwndms0812 commented:
> 저기 위에 부분 말고도 jwtautorization 파일에 반영이 안된 부분 하나가 있었는데(해결 된 것 확인), 일단 merge 해주시면 제가 jwtproperties랑 필요없는 주석 삭제, 본문에 작성한 부분 반영 하도록 하겠습니다
> 
> pr대신 날려주셔서 감사해요1

Commit:
feat: properties 파일을 이용한 환경변수 사용 구현

- .gitignore 파일 루트로 이동
- EnvConfiguraion 파일 생성
- JwtAuthenticationFilter, JwtAuthorizedFilter에 환경변수 적용
- JwtProperties 파일 이력 삭제

owel-dev commented:
하는 김에 JwtProperties.java 파일 이력 없애고
env.properties 파일 이용하여 환경변수 사용하는 기능 추가하였습니다.
관련해서 참고한 내용들 노션 학습자료공유에 추가했습니다!

dlwndms0812 commented:
넵 확인했습니다
